File: MDC/MDC.pl
mdc(X, 0, X):- !.
mdc(0, X, X):- !.
mdc(X, Y, D):- X > Y, !, Z is X mod Y, mdc(Y, Z, D).
mdc(X, Y, D):- Z is Y mod X, mdc(X, Z, D).

File: MDC/MDC.c
#include<stdio.h>
int MDC(int u, int v) {
  if (u < 0) u = -u;
  if (v < 0) v = -v;
  if (v) while ((u %= v) && (v %= u));
  return (u + v);
}

void main(){
int primeiroNum, segundoNum;

scanf("%d", &primeiroNum);
scanf("%d", &segundoNum);
int resultado = MDC(primeiroNum, segundoNum);
printf("%d", resultado);
}
